src/middlewared/middlewared/service/core_service.py

```diff
@@ -101,6 +101,11 @@ def __job_by_credential_and_id(self, credential, job_id):
 
         job = self.middleware.jobs[job_id]
 
+        if job.credentials is None:
+            raise CallError(
+                'Only users with full administrative privileges can download internal job logs', errno.EPERM,
```

Comment:
Can't someone hit this by running `core.job_wait` for any job that's owned by root?

Reply:
You are correct. Fixed the error message.